internal/mapper/datasource_mapper_test.go

```diff
@@ -722,18 +722,20 @@ func TestDataSourceMapper_basic_merges(t *testing.T) {
 					"nested_map": base.CreateSchemaProxy(&base.Schema{
 						Type:        []string{"object"},
 						Description: "hey this is a map!",
-						AdditionalProperties: base.CreateSchemaProxy(&base.Schema{
-							Type: []string{"object"},
-							Properties: map[string]*base.SchemaProxy{
-								"deep_nested_bool": base.CreateSchemaProxy(&base.Schema{
-									Type: []string{"boolean"},
-								}),
-								"deep_nested_int64": base.CreateSchemaProxy(&base.Schema{
-									Type:        []string{"integer"},
-									Description: "hey this is an int64!",
-								}),
-							},
-						}),
+						AdditionalProperties: &base.DynamicValue[*base.SchemaProxy, bool]{
```

Comment:
The major breaking change was the structure of the `AdditionalProperties` field, which has been changed from an `any` to a strongly typed value! 🥳